ethers-middleware/src/signer.rs

```diff
@@ -251,6 +252,24 @@ where
             tx.set_chain_id(chain_id);
         }
 
+        // If a chain_id is matched to a known chain that doesn't support EIP-1559, automatically
+        // change transaction to be Legacy type.
+        if tx.chain_id().is_some() {
+            let tx_chain_id = tx.chain_id().unwrap(); // safe unwrap after is_some check above
+            let chain = Chain::try_from(tx_chain_id.as_u64());
+            if chain.is_ok() && chain.unwrap().is_legacy() {
+                // safe unwrap after is_ok check
+                let legacy_tx = match tx.clone() {
+                    TypedTransaction::Eip1559(inner) => {
+                        let i_tx: TransactionRequest = inner.into();
+                        TypedTransaction::Legacy(i_tx)
+                    }
+                    other => other,
+                };
+                *tx = legacy_tx;
```

Comment:
this clone is unnecessary if already legacy.
match via `ref` should work here and clone if eip1559